tests/Query/TermsTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Solr Client Symfony package.
+ *
+ * (c) ingatlan.com Zrt. <fejlesztes@ingatlan.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace iCom\SolrClient\Tests\Query;
+
+use iCom\SolrClient\Query\Terms;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \iCom\SolrClient\Query\Terms
+ */
+final class TermsTest extends TestCase
+{
+    /** @test */
+    public function it_throws_exception_on_invalid_method(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+
+        Terms::create('id', [])->method('method');
+    }
+
+    /**
+     * @test
+     * @dataProvider provider
+     */
+    public function it_creates_terms_query_string(\Closure $terms, string $expectedTerms): void
+    {
+        $this->assertSame($expectedTerms, (string) $terms());
+    }
+
+    public function provider(): iterable
+    {
+        yield 'field' => [
+            static function (): Terms { return Terms::create('id', []); },
+            '{!terms f=id}',
+        ];
+
+        yield 'method' => [
+            static function (): Terms { return Terms::create('id', [])->method('termsFilter'); },
+            '{!terms f=id method=termsFilter}',
+        ];
+
+        yield 'separator-comma' => [
+            static function (): Terms { return Terms::create('id', [1, 2, 3])->separator(','); },
+            '{!terms f=id separator=","}1,2,3',
+        ];
+
+        yield 'separator-space' => [
+            static function (): Terms { return Terms::create('id', [1, 2, 3])->separator(' '); },
+            '{!terms f=id separator=" "}1 2 3',
+        ];
+
+        yield 'separator-double-quote' => [
+            static function (): Terms { return Terms::create('id', [1, 2, 3])->separator('"'); },
+            '{!terms f=id separator="\""}1"2"3',
+        ];
+
+        yield 'separator-single-quote' => [
+            static function (): Terms { return Terms::create('id', [1, 2, 3])->separator('\''); },
```

Comment:
Itt mért van pluszban `\` a `'` előtt? Nem baj, ha van olyan teszt is, hogy `\`-is van benne, de akkor a nevét írjuk át.

Reply:
Itt ennyire tudtam megszépíteni a query-t:` '{!terms f=id separator="\\\'"}'."1'2'3"'` a slashek kellenek továbbra is  a separatornál. A solr végül ezt fogja kapni: `{!terms f=id separator="\'"}1'2'3` így is jól működik.